File: Binary_Search.cpp
// C++ program to implement recursive Binary Search

#include<iostream>
using namespace std;

// A recursive binary search function. It returns
// location of x in given array arr[l..r] is present,
// otherwise -1
class Solution {
    public:
    int searchsortedarray(int arr[],int n,int k)
    {
        int low=0,high=n-1;
        while(low<=high)
        {
            int mid =(low+high)/2;

            // If the element is present at the middle
            // itself
            if(arr[mid]==k)
                return 1;

             // If element is smaller than mid, then
            // it can only be present in left subarray
            else if(arr[mid]>k)
                high=mid-1;

            // Else the element can only be present
            // in right subarray
            else
                low=mid+1;
        }
        return -1;
    }
};

int main()
{
    int size;
    cout<<"Enter the size of the array : ";
    cin>>size;
    int arr[size];
    cout<<"Enter the elements of the array : ";
    for(int i=0;i<size;i++)
        cin>>arr[i];
    int target;
    cout<<"Enter the element to be searched: ";
    cin>>target;
    Solution ans;
    int val=ans.searchsortedarray(arr,size,target);
    if(val>0)
    {
        cout<<"The element is present  ";
    }
    else 
        cout<<"Element is not found";
    return 0;
}

File: BinarySearch.java
package Lab10;

public class BinarySearch {
    public static int binarySearch(int[] array,int low,int high,int key){
            int mid = (low+high)/ 2;
            if (high<low){return -1;}
           else if (array[mid] == key) {
                return mid;
            }
            else if ( key<array[mid]) {
                return binarySearch(array, low, mid - 1, key);
            }
            else if (key>array[mid]){
                return binarySearch(array, low+1, mid , key);
            }
return -1;

    }
    public static void main(String[] args) {
        int[] array={1,2,3,4,5,6,8};
int last= array.length-1;
int result=binarySearch(array,0,last,4);
if(result==-1){
    System.out.println("Not found");
}
else {
    System.out.println("Element is found at index :"+result);
}
    }
}
